### PR TITLE
Jitsi Widget Permissions

### LINKED ISSUE #3391: Widget Permission for jitsi widgets
Notice that currently display name / avatar are not passed
Also
> it should default to jitsi.riot.im if the widget's data is missing a domain
but if the data has a domain, it should use that one

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3391

Depends on https://github.com/matrix-org/matrix-android-sdk/pull/514

### Pull Request Checklist
image

* [ ] Changes has been tested on an Android device or Android emulator with API 16
* [x] UI change has been tested on both light and dark themes
* [x] Pull request is based on the develop branch
* [x] Pull request updates [CHANGES.rst](https://github.com/vector-im/riot-android/blob/develop/CHANGES.rst)
* [x] Pull request includes screenshots or videos if containing UI changes

### CODE REVIEW

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 1]
```diff
@@ -1765,10 +1768,31 @@ private void launchJitsiActivity(Widget widget, boolean aIsVideoCall) {
...
+                bs.setOnFinish((accepted) -> {
+                    if (accepted) {
+                        final Intent intent = new Intent(this, JitsiCallActivity.class);
+                        intent.putExtra(JitsiCallActivity.EXTRA_WIDGET_ID, widget);
+                        intent.putExtra(JitsiCallActivity.EXTRA_ENABLE_VIDEO, aIsVideoCall);
+                        startActivity(intent);
+                    }
+                    return Unit.INSTANCE;
+                });
+                bs.show(getSupportFragmentManager(), "JitsyPerm");
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: Typo: 'JitsiPerm' with a "i" :)

File: vector/src/main/java/im/vector/activity/WidgetActivity.kt [hunk 2]
```diff
@@ -65,7 +65,11 @@ class WidgetActivity : VectorAppCompatActivity() {
                         //already there
                     } else {
                         RoomWidgetPermissionBottomSheet
-                                .newInstance(viewModel.session!!.myUserId, viewModel.widget)
+                                .newInstance(viewModel.session!!.myUserId, viewModel.widget).apply {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/WidgetActivity.kt]: Does this code is robust to screen rotation, will the bottomSheet is displayed?

File: vector/src/main/java/im/vector/fragments/roomwidgets/RoomWidgetPermissionBottomSheet.kt [hunk 3]
```diff
@@ -53,8 +52,8 @@ class RoomWidgetPermissionBottomSheet : VectorBaseBottomSheetDialogFragment() {
 
     @BindView(R.id.bottom_sheet_widget_permission_owner_avatar)
     lateinit var authorAvatarView: ImageView
-
-    private val sharedActivityViewModel: RoomWidgetViewModel by activityViewModel()
+    
+    var onFinish: ((Boolean) -> Unit)? = null
```
Comment (bmarty) [inline vector/src/main/java/im/vector/fragments/roomwidgets/RoomWidgetPermissionBottomSheet.kt]: The sharedViewModel was better, no?

Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/fragments/roomwidgets/RoomWidgetPermissionBottomSheet.kt]: It makes the bottom sheet more generic, can be used in other activity easier

File: vector/src/main/java/im/vector/activity/WidgetActivity.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/activity/WidgetActivity.kt]: worst case you have to tap again on the video conf banner

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 4]
```diff
@@ -1772,7 +1772,7 @@ private void launchJitsiActivity(Widget widget, boolean aIsVideoCall) {
 
             String domain = UrlUtilKt.extractDomain(JitsiCallActivity.JITSI_SERVER_URL);
             if (domain == null) return; //display a toast?
-            boolean isAllowed = mSession.getIntegrationManager().isVideoConfDomainAllowed(domain);
+            boolean isAllowed = mSession.getIntegrationManager().isNativeWidgetAllowed("jitsi", domain);
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: 👍

**Pull request merged**